contracts/casino/src/casino.cpp

```diff
@@ -163,4 +166,82 @@ uint32_t casino::get_profit_margin(uint64_t game_id) const {
     return platform::read::get_game(get_platform(), game_id).profit_margin;
 }
 
+void casino::set_bonus_admin(name new_admin) {
+    require_auth(get_owner());
+    check(is_account(new_admin), "new bonus admin account does not exist");
+    bstate.admin = new_admin;
+}
+
+void casino::deposit_bonus(asset quantity, const std::string& memo) {
+    require_auth(bstate.admin);
+    check(memo.size() <= 256, "memo has more than 256 bytes");
+    bstate.total_allocated += quantity;
+    check(bstate.total_allocated <= token::get_balance(get_self(), core_symbol), "total bonus cannot exceed casino balance");
+}
+
+void casino::withdraw_bonus(name to, asset quantity, const std::string& memo) {
```

Comment:
How that affects to casino liquidity pool? Bonus pool should be strictly separated from liquidity pool, for now it's possible to withdraw funds from liquidity pool bypassing all withdrawal limits peresented here https://github.com/DaoCasino/platform-contracts/blob/master/contracts/casino/src/casino.cpp#L116
PoC: deposit funds to liquidity pool -> deposit_bonus -> withdraw_bonus -> PROFIT! (withdrawal bypassing liq. pool withdrawal limits)